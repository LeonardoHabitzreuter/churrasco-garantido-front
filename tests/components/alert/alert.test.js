import React from 'react'
import { shallow } from 'enzyme'
import Alert from '../../../src/components/alert'
import { Alert as AntAlert } from 'antd'

describe('alert component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<Alert show={false} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should appear only if the show prop equals true', () => {
    const wrapper = shallow(<Alert show={false} />)
    expect(wrapper.find(AntAlert)).toHaveLength(0)
    wrapper.setProps({ show: true })
    expect(wrapper.find(AntAlert)).toHaveLength(1)
  })

  test('should be able to render more than one message', () => {
    const wrapper = shallow(<Alert show messages={['test1', 'test2']} />)
    expect(wrapper.find(AntAlert).prop('description').map(pTag => pTag.props.children)).toEqual(expect.arrayContaining(['test1', 'test2']))
  })
})
