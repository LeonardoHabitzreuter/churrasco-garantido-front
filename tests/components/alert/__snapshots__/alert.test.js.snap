// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert component should render correctly 1`] = `""`;
